components/views.py

```diff
@@ -1,67 +1,63 @@
 from django.core.paginator import EmptyPage, PageNotAnInteger, Paginator
-from django.shortcuts import get_object_or_404
+from django_filters import rest_framework as filters
 from rest_framework import generics, status
 from rest_framework.response import Response
-from rest_framework.views import APIView
 
-from .filters import ComponentFilter
-from .models import Component
-from .serializers import ComponentListSerializer, ComponentSerializer
+from blueprintapi.filters import ObjectPermissionsFilter
+from components.filters import ComponentFilter
+from components.models import Component
+from components.serializers import ComponentListSerializer, ComponentSerializer
 
 
 class ComponentListView(generics.ListCreateAPIView):
     """
     Use for read-write endpoints to represent a collection of model instances.
     Provides get and post method handlers.
     """
-
     queryset = Component.objects.all().order_by("pk")
     serializer_class = ComponentListSerializer
-
-    def list(self, request):
-        queryset = self.get_queryset()
-        serializer = ComponentListSerializer(queryset, many=True)
-        return Response(serializer.data, status=status.HTTP_200_OK)
+    filter_backends = [ObjectPermissionsFilter, ]
```

Comment:
For list views, we need to be sure to add this permissions filter: only allowed objects should be returned for list views.